apps/root/src/hooks/useTransactionsHistory.tsx

```diff
@@ -82,10 +87,26 @@ function useTransactionsHistory(): {
         resolvedEvents.splice(insertionIndex, 0, tx);
       });
 
+      let resolvedFilteredEvents: TransactionEvent[] = [];
+
+      if (!!tokens) {
+        // For token history, return only token-related events
+        resolvedFilteredEvents = resolvedEvents.filter(
+          (tx) =>
+            !tokens ||
+            getTransactionInvolvedTokens(tx)
+              .map((token) => getTokenListId({ chainId: token.chainId, tokenAddress: token.address }))
+              .some((tokenId) => tokens.includes(tokenId))
+        );
+      } else if (lastEventTimestamp) {
```

Comment:
what about the default case?

Reply:
If lastEventTimestamp doesn't exist, data was never fetched for global events (It should never happen in the use of this hook, thanks to promises initializer). So the empty list should be fine... What do you think?